src/v2/Components/Onboarding/Steps/Budget.tsx

```diff
@@ -27,98 +47,75 @@ const OptionsContainer = styled.div`
   `};
 `
 
-interface State {
-  selection: number | null
+const budgetOptions = {
+  "UNDER $500": 500,
+  "UNDER $2,500": 2500,
+  "UNDER $5,000": 5000,
+  "UNDER $10,000": 10000,
+  "UNDER $25,000": 25000,
+  "UNDER $50,000": 50000,
+  "NO BUDGET IN MIND": 1000000000000,
 }
 
-@track({}, { dispatch: data => Events.postEvent(data) })
-export class BudgetComponent extends React.Component<
-  Props & StepProps & SystemContextProps,
-  State
-> {
-  options = {
-    "UNDER $500": 500,
-    "UNDER $2,500": 2500,
-    "UNDER $5,000": 5000,
-    "UNDER $10,000": 10000,
-    "UNDER $25,000": 25000,
-    "UNDER $50,000": 50000,
-    "NO BUDGET IN MIND": 1000000000000,
-  }
+export const BudgetComponent = props => {
+  const tracking = useTracking()
+  const updateProfile = props.updateUserProfile || updateUserProfile
+  const [selectedOption, setSelectedOption] = useState(null)
 
-  state = {
-    selection: null,
+  const onOptionSelected = (index: number) => {
+    const selection = { selection: Object.values(budgetOptions)[index] }
+    setSelectedOption(selection)
   }
 
-  onOptionSelected = (index: number) => {
-    const selection = { selection: Object.values(this.options)[index] }
-    this.setState(selection)
-  }
+  const options = Object.keys(budgetOptions).map((text, index) => (
+    <SelectableToggle
+      key={index}
+      text={text}
+      onSelect={() => onOptionSelected(index)}
+      selected={selectedOption?.selection === budgetOptions[text]}
+    />
+  ))
 
-  submit() {
-    const priceRangeMax = this.state.selection
+  const submit = () => {
+    const priceRangeMax = selectedOption?.selection
+    if (!priceRangeMax) return
 
-    commitMutation<BudgetUpdateMyUserProfileMutation>(
-      this.props.relayEnvironment,
-      {
-        // TODO: Inputs to the mutation might have changed case of the keys!
-        mutation: graphql`
-          mutation BudgetUpdateMyUserProfileMutation(
-            $input: UpdateMyProfileInput!
-          ) {
-            updateMyUserProfile(input: $input) {
-              user {
-                name
-              }
-            }
-          }
-        `,
-        variables: {
-          input: {
-            priceRangeMin: -1,
-            priceRangeMax,
-          },
-        },
-      }
-    )
+    updateProfile(priceRangeMax, props.relayEnvironment)
 
-    const redirectTo = this.props.redirectTo || "/"
+    const redirectTo = props.redirectTo || "/"
     setTimeout(() => window.location.assign(redirectTo), 500)
 
-    this.props.tracking.trackEvent({
+    const event = {
+      context_module: ContextModule.onboardingInterests,
+      context_owner_type: OwnerType.onboarding,
+      data_input: selectedOption,
+    }
+    tracking.trackEvent(event)
+
+    const completedEvent: any = {
```

Comment:
Does this need to be an `any`?

Reply:
Sadly I believe it does. This is a legacy event that I discussed with @abhitip and we decided to just leave it be. This is the type check error I get if I remove the any:

```
src/v2/Components/Onboarding/Steps/Budget.tsx:104:25 - error TS2345: Argument of type '{ action: string; }' is not assignable to parameter of type 'Partial<AuctionInfo> | Partial<ContextModule> | Partial<ContextPage> | Partial<Flow> | ... 70 more ... | Partial<...>'.
  Type '{ action: string; }' is not assignable to type 'Partial<ToggledNotification>'.
    Types of property 'action' are incompatible.
      Type 'string' is not assignable to type 'ActionType.toggledNotification'.

104     tracking.trackEvent(completedEvent)
                            ~~~~~~~~~~~~~~
```

Open to other ideas but this felt not worth pursuing. 🤷